### PR TITLE
Dshield feed integration

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Dshield feed integration.

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [x] 5.5.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by roysagi** [comment]
still missing playbook

File: Packs/FeedDshield/Integrations/FeedDshield/FeedDshield.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        "regex": r"^(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})\t[\d.]*\t(\d{1,2})",
+        "transform": "\\1/\\2"
+    })
+
+    fields = json.dumps({
+        "dshield_nattacks": {
+            "regex": "^.*\\t.*\\t[0-9]+\\t([0-9]+)",
+            "transform": "\\1"
+        },
+        # name will change.
```
Comment (ronykoz) [inline Packs/FeedDshield/Integrations/FeedDshield/FeedDshield.py]: why

Comment (roysagi) [reply Packs/FeedDshield/Integrations/FeedDshield/FeedDshield.py]: @David-BMS - needed to decide on a name - eventually decided to take out the field altogether

**Comment by roysagi** [comment]
image

File: Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            "regex": "^.*\\t.*\\t[0-9]+\\t[0-9]+\\t[^\\t]+\\t([A-Z]+)",
+            "transform": "\\1"
+        },
+        "registrarabuseemail": {
+            "regex": "^.*\\t.*\\t[0-9]+\\t[0-9]+\\t[^\\t]+\\t[A-Z]+\\t(\\S+)",
+            "transform": "\\1"
+        }
+    })
+    params['fields'] = fields
+
```
Comment (avidan-H) [inline Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py]: You haven't added the `custom_fields_mapping` parameter to `params` that you added to the httpapimodule in your other PR.

Comment (ronykoz) [inline Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py]: saw that you've deleted some fields from here, why?

Comment (roysagi) [reply Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py]: once the other PR will be merged I'll add this.

Comment (roysagi) [reply Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py]: talking to @David-BMS - we decided these fields are not relevant. So I took them off.

Comment (ronykoz) [reply Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py]: it sounds like minemeld parsed them so we should too, lets add them back.
Users will decide what to do with them on the mapping front

Comment (ronykoz) [reply Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py]: lets add this now

File: Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
+category: Data Enrichment & Threat Intelligence
+commonfields:
+  id: DShield Feed
+  version: -1
+configuration:
+- display: Fetch indicators
+  name: feed
```
Comment (avidan-H) [inline Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.yml]:
```suggestion
  name: feed
  defaultvalue: true
```

File: Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py (hunk 2 above)
Comment (roysagi) [reply Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py]: added

Comment (roysagi) [reply Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py]: done

**Comment by roysagi** [comment]
Hi @kirbles19 if you can please review the [yml file](https://github.com/demisto/content/blob/dshield_feed/Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.yml)

**Comment by roysagi** [comment]
All fields in the yml are default and @kirbles19 approved the TIP excel so I'm adding docs approved

File: Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        "regex": r"^(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})\t[\d.]*\t(\d{1,2})",
+        "transform": "\\1/\\2"
+    })
+
+    fields = json.dumps({
+        "numberofattacks": {
+            "regex": "^.*\\t.*\\t[0-9]+\\t([0-9]+)",
+            "transform": "\\1"
+        },
+        "networkneame": {
```
Comment (avidan-H) [inline Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py]:
```suggestion
        "networkname": {
```

File: Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    params['indicator_type'] = FeedIndicatorType.CIDR
+
+    params['url'] = 'https://www.dshield.org/block.txt'
+    params['ignore_regex'] = "[#S].*"
+    params['indicator'] = json.dumps({
+        "regex": r"^(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})\t[\d.]*\t(\d{1,2})",
+        "transform": "\\1/\\2"
+    })
+
+    fields = json.dumps({
```
Comment (avidan-H) [inline Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py]: Why `json.dumps`?

Comment (roysagi) [reply Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py]: only way I was able to make it work - str changes it to single quotation mark which creates an error.

File: Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py (hunk 4 above)
Comment (roysagi) [reply Packs/FeedDShield/Integrations/FeedDShield/FeedDShield.py]: fixed

File: Packs/FeedDShield/TestPlaybooks/playbook-DshieldFeed_test.yml [hunk 6]
```diff
@@ -0,0 +1,184 @@
+elasticcommonfields: {}
+id: playbook-DshieldFeed_test
+fromversion: 5.5.0
+version: 3
```
Comment (avidan-H) [inline Packs/FeedDShield/TestPlaybooks/playbook-DshieldFeed_test.yml]:
```suggestion
version: -1
```

File: Packs/FeedDShield/TestPlaybooks/playbook-DshieldFeed_test.yml [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+      "dimensions": {
+        "height": 825,
+        "width": 380,
+        "x": 50,
+        "y": 50
+      }
+    }
+  }
+inputs: []
+outputs: []
```
Comment (avidan-H) [inline Packs/FeedDShield/TestPlaybooks/playbook-DshieldFeed_test.yml]:
```suggestion
outputs: []
fromversion: 5.5.0
```

**Pull request merged**